src/ctelnet.cpp

```diff
@@ -1804,18 +1804,61 @@ void cTelnet::processTelnetCommand(const std::string& telnetCommand)
                     cmd += TN_SB;
                     cmd += OPT_TERMINAL_TYPE;
                     cmd += TNSB_IS;
-                    /*
-                     * The valid characters for termTerm are more restricted
-                     * than being ASCII - from:
-                     * https://tools.ietf.org/html/rfc1010 (page 29):
-                     * "A terminal names may be up to 40 characters taken from
-                     * the set of uppercase letters, digits, and the two
-                     * punctuation characters hyphen and slash.  It must start
-                     * with a letter, and end with a letter or digit."
-                     * Once we comply with that we can be certain that Mud
-                     * Server encoding will NOT be an issue!
-                     */
-                    cmd += termType.toUtf8().constData();
+
+                    switch (mCycleCountMTTS) {
+                        case 0:
+                            /*
+                            * The valid characters for termTerm are more restricted
+                            * than being ASCII - from:
+                            * https://tools.ietf.org/html/rfc1010 (page 29):
+                            * "A terminal names may be up to 40 characters taken from
+                            * the set of uppercase letters, digits, and the two
+                            * punctuation characters hyphen and slash.  It must start
+                            * with a letter, and end with a letter or digit."
+                            * Once we comply with that we can be certain that Mud
+                            * Server encoding will NOT be an issue!
+                            */
+                            cmd += termType.replace(QChar::Space, "-").replace(".", "-").toUtf8().constData();
```

Comment:
I *think* this will completely "sanitise" the version string to meet the above restrictions - it is intended to:
* make the string UPPER CASE
* convert periods (like those in the current `4.17.2` value) to `.`
* convert spaces to hyphens (so there is no longer a space between "MUDLET" and what follows) - hopefully these three changes won't mess up existing Game Servers :worried: 
* change any other character (which are NOT allowed) to hyphens
* convert any multiple adjacent hyphens to a single one
* limit the string to 40 characters
* removes the end character repeatedly if it is a not a letter or a number 
* Crashes debug builds of Mudlet should the version string end up as an empty one! :scream: 
```suggestion
                            static const auto allInvalidCharacters = QRegularExpression(qsl("[^A-Z,0-9,-,\\/]"));
                            static const auto multipleHyphens = QRegularExpression(qsl("-{2,}"));
                            QString sanitisedTermType = termType.toUpper()
                                                                .replace(QChar('.'), QChar('/'))
                                                                .replace(QChar::Space, QChar('-'))
                                                                .replace(allInvalidCharacters, QChar('-'))
                                                                .replace(multipleHyphens, QChar('-'))
                                                                .left(40)
                                                            .toLatin1().constData();
                            while (!sanitisedTermType.isEmpty() && !sanitisedTermType.back().isLetterOrNumber()) {
                                sanitisedTermType.chop(1);
                            }
                            Q_ASSERT_X(!sanitisedTermType.isEmpty(),
                                       "cTelnet::processTelnetCommand(...)",
                                       "ended up with an empty version string whilst trying to sanitise the Mudlet one");
                            cmd += sanitisedTermType.toLatin1().constData();
```

Reply:
Added